webpack/ForemanTasks/Components/TasksTable/TasksBulkActions.js

```diff
@@ -161,3 +168,84 @@ export const bulkCancelById = ({
     }
   }
 };
+
+const handleErrorForceCancel = (error, dispatch) => {
+  dispatch({ type: TASKS_FORCE_CANCEL_FAILURE, error });
+  dispatch(
+    addToast({
+      type: TOAST_TYPES.ERROR,
+      message: `${__(`Cannot force cancel tasks at the moment`)} ${error}`,
+    })
+  );
+};
+
+export const bulkForceCancelById = ({
+  selected,
+  url,
+  parentTaskID,
+}) => async dispatch => {
+  const stopTasks = selected.filter(task => task.state !== 'stopped');
+  if (stopTasks.length < selected.length)
+    dispatch(
+      addToast({
+        type: TOAST_TYPES.WARNING,
+        message: sprintf(
+          '%s task(s) are already stopped',
+          selected.length - stopTasks.length
+        ),
+      })
+    );
+  if (stopTasks.length > 0) {
+    dispatch({ type: TASKS_FORCE_CANCEL_REQUEST });
+    try {
+      const { data } = await bulkByIdRequest(stopTasks, BULK_FORCE_CANCEL_PATH);
+      dispatch({ type: TASKS_FORCE_CANCEL_SUCCESS });
+      if (data.stopped_length) {
+        dispatch(
+          addToast({
+            type: TOAST_TYPES.SUCCESS,
+            message: sprintf(
+              '%s task(s) cancelled with force',
+              data.stopped_length
+            ),
+          })
+        );
+      }
+      if (data.skipped_length > 0)
+        dispatch(
+          addToast({
+            type: TOAST_TYPES.WARNING,
+            message: sprintf(
+              '%s task(s) are already stopped',
+              data.skipped_length
+            ),
+          })
+        );
+      if (data.stopped_length > 0) {
+        reloadPage(url, parentTaskID, dispatch);
+      }
+    } catch (error) {
+      handleErrorForceCancel(error, dispatch);
+    }
+  }
+};
+
+export const bulkForceCancelBySearch = ({
+  query,
+  parentTaskID,
+}) => async dispatch => {
+  dispatch({ type: TASKS_FORCE_CANCEL_REQUEST });
+  dispatch(
+    addToast({
+      type: 'info',
```

Comment:
can you reuse the `TOASTS_TYPES.INFO` constant ?
and btw, didn't you created some helper to make the `addToast` action shorter? e.g: `addInfoToast(message)` ?

Reply:
Yeah, changed the helper so it would be more generic